src/js/actions/layers.js

```diff
@@ -758,6 +776,56 @@ define(function (require, exports) {
             return Promise.resolve();
         }
 
+        if (descendants) {
+            layers = layers.flatMap(document.layers.descendants, document.layers);
+        }
+
+        // When collapsing layers, if a visible descendent is selected then the selection
```

Comment:
would it save a few cycles to only do all this selection logic ONLY if `!expand`?